cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -7,65 +7,528 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
+	"fmt"
+	"regexp"
+	"strings"
 
+	"github.com/azure/azure-dev/cli/azd/pkg/azdo"
 	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
 	"github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
 	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
 	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/git"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/build"
+	azdoGit "github.com/microsoft/azure-devops-go-api/azuredevops/git"
 )
 
-// AzdoHubScmProvider implements ScmProvider using Azure DevOps as the provider
+// AzdoScmProvider implements ScmProvider using Azure DevOps as the provider
 // for source control manager.
-type AzdoHubScmProvider struct {
+type AzdoScmProvider struct {
+	repoDetails    *AzdoRepositoryDetails
```

Comment:
Is there a reason this is pointer instead of value?  If it was a value, I think that we might be able to remove getRepoDetails since creating a new `AzdoScmProvider` instance would end up allocating the zero value for this type (which is all getRepoDetails is doing)?

Reply:
https://github.com/Azure/azure-dev/issues/756